nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/ProcessorResource.java

```diff
@@ -423,6 +429,11 @@ public Response getPropertyDescriptor(
         // get the property descriptor
         final PropertyDescriptorDTO descriptor = serviceFacade.getProcessorPropertyDescriptor(id, propertyName);
 
+        // Adjust sensitive status for dynamic properties based on requested status
+        if (descriptor.isDynamic()) {
+            descriptor.setSensitive(sensitive);
+        }
```

Comment:
I don't think this is correct. The sensitive flag is, by default, false. There may be processors that already have set a particular dynamic property to sensitive, and if the UI requests that property but doesn't include a value for the 'sensitive' flag, even though the processor itself says it's sensitive, this would mark it non-sensitive. I think we need to change this to:
```
if (descriptor.isDynamic() && sensitive) {
    descriptor.setSensitive(true);
}
```